src/libtorchaudio/rnnt/cpu/compute.cpp

```diff
@@ -1,77 +1,89 @@
-  TORCH_CHECK(targets.dtype() == torch::kInt32, "targets must be int32 type");
-  TORCH_CHECK(
-      logit_lengths.dtype() == torch::kInt32,
+
+  STD_TORCH_CHECK(
+      targets.scalar_type() == ScalarType::Int, "targets must be int32 type");
+
+  STD_TORCH_CHECK(
+      logit_lengths.scalar_type() == ScalarType::Int,
       "logit_lengths must be int32 type");
-  TORCH_CHECK(
-      target_lengths.dtype() == torch::kInt32,
+  STD_TORCH_CHECK(
+      target_lengths.scalar_type() == ScalarType::Int,
       "target_lengths must be int32 type");
 
-  TORCH_CHECK(logits.is_contiguous(), "logits must be contiguous");
-  TORCH_CHECK(targets.is_contiguous(), "targets must be contiguous");
-  TORCH_CHECK(
+  STD_TORCH_CHECK(logits.is_contiguous(), "logits must be contiguous");
+  STD_TORCH_CHECK(targets.is_contiguous(), "targets must be contiguous");
+  STD_TORCH_CHECK(
       logit_lengths.is_contiguous(), "logit_lengths must be contiguous");
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       target_lengths.is_contiguous(), "target_lengths must be contiguous");
 
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       logits.dim() == 4, "logits must be 4-D (batch, time, target, class)");
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       targets.dim() == 2, "targets must be 2-D (batch, max target length)");
-  TORCH_CHECK(logit_lengths.dim() == 1, "logit_lengths must be 1-D");
-  TORCH_CHECK(target_lengths.dim() == 1, "target_lengths must be 1-D");
+  STD_TORCH_CHECK(logit_lengths.dim() == 1, "logit_lengths must be 1-D");
+  STD_TORCH_CHECK(target_lengths.dim() == 1, "target_lengths must be 1-D");
 
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       logit_lengths.size(0) == logits.size(0),
       "batch dimension mismatch between logits and logit_lengths");
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       target_lengths.size(0) == logits.size(0),
       "batch dimension mismatch between logits and target_lengths");
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       targets.size(0) == logits.size(0),
       "batch dimension mismatch between logits and targets");
 
-  TORCH_CHECK(
+  STD_TORCH_CHECK(
       blank >= 0 && blank < logits.size(-1),
       "blank must be within [0, logits.shape[-1])");
 
-  TORCH_CHECK(
-      logits.size(1) == at::max(logit_lengths).item().toInt(),
-      "input length mismatch");
-  TORCH_CHECK(
-      logits.size(2) == at::max(target_lengths).item().toInt() + 1,
+  auto max_ivalue = [](const Tensor& t) {
+    return reinterpret_cast<int32_t*>(torch::stable::amax(t, {}).data_ptr())[0];
+  };
+
+  STD_TORCH_CHECK(
+      logits.size(1) == max_ivalue(logit_lengths), "input length mismatch");
+  STD_TORCH_CHECK(
+      logits.size(2) == max_ivalue(target_lengths) + 1,
       "output length mismatch");
-  TORCH_CHECK(
-      targets.size(1) == at::max(target_lengths).item().toInt(),
-      "target length mismatch");
+  STD_TORCH_CHECK(
+      targets.size(1) + 1 == logits.size(2), "target length mismatch");
 
   Options options;
```

Comment:
Nit, we can leave it as a TODO as this isn't something introduced in this PR anyway, but worth knowing for later if we ever clean that up:

```suggestion
  // TODO: Use static_cast and check bounds when down-casting from double to float (clamp_)
  // and from int64 to int (blank_)
  Options options;
```

Reply:
Applied, thanks!